patches/tModLoader/Terraria/Social/Steam/WorkshopHelper.TML.cs

```diff
@@ -459,41 +478,61 @@ internal void WaitForQueryResult(SteamAPICall_t query)
 
 			/////// Process Query Result per Item ////////////////////
 
-			//TODO: This Method and it's downstream callers needs work to remove default passed values. Deferred during PR #3346
-			internal ModDownloadItem GenerateModDownloadItemFromQuery(uint i)
+			internal WorkshopSearchReturnState TryGenerateModDownloadItem(uint index, out ModDownloadItem item)
+			{
+				item = null;
+				try {
+					item = GenerateModDownloadItemFromQuery(index);
+					return WorkshopSearchReturnState.Success;
+				}
+				catch (BannedModException e) {
+					item = new ModDownloadItem(e.displayName, e.modPubId, banned: true);
+					return WorkshopSearchReturnState.Banned;
+				}
+				catch (Exception) {
+					return WorkshopSearchReturnState.RetrievalFailed;
+				}
+			}
+
+			private ModDownloadItem GenerateModDownloadItemFromQuery(uint i)
 			{
 				// Item Result call data
 				SteamUGCDetails_t pDetails = SteamedWraps.FetchItemDetails(_primaryUGCHandle, i);
 
 				PublishedFileId_t id = pDetails.m_nPublishedFileId;
 
 				if (pDetails.m_eResult != EResult.k_EResultOK) {
-					Logging.tML.Warn("Unable to fetch mod PublishId#" + id + " information. " + pDetails.m_eResult);
-					return null;
+					throw new SocialBrowserException("Unable to fetch mod PublishId#" + id + " information. " + pDetails.m_eResult);
 				}
 
 				string ownerId = pDetails.m_ulSteamIDOwner.ToString();
 
 				DateTime lastUpdate = Utils.UnixTimeStampToDateTime((long)pDetails.m_rtimeUpdated);
 				string displayname = pDetails.m_rgchTitle;
+				bool banned = pDetails.m_bBanned;
 
-				// Item Tagged data
+				if (banned) {
+					throw new BannedModException($"Item {id}: '{displayname}' is banned. Skipping...", displayname, id.ToString());
```

Comment:
No need to say `Skipping...`, in fact it's not really this method's job to say what's going to happen with the exception when it's handled